php/admin/class-onboarding.php

```diff
@@ -0,0 +1,392 @@
+			array(),
+			$this->plugin->get_version()
+		);
+	}
+
+	/**
+	 * Render the Welcome message.
+	 */
+	public function show_welcome_notice() {
+		$example_post_id = get_option( $this->option );
+
+		if ( ! $example_post_id ) {
+			return;
+		}
+		?>
+		<div class="block-lab-welcome block-lab-notice notice is-dismissible">
+			<h2>🖖 <?php esc_html_e( 'Welcome, traveller!', 'block-lab' ); ?></h2>
+			<p class="intro"><?php esc_html_e( 'Block Lab makes it super easy to build custom blocks for the WordPress editor.', 'block-lab' ); ?></p>
+			<p><strong><?php esc_html_e( 'Want to see how it\'s done?', 'block-lab' ); ?></strong> <?php esc_html_e( 'Here\'s one I prepared earlier.', 'block-lab' ); ?></p>
+			<?php
+			edit_post_link(
+				__( 'Let\'s get started!', 'block-lab' ),
+				'',
+				'',
+				$example_post_id,
+				'button button--white button_cta'
+			);
+			?>
+			<p class="ps"><?php esc_html_e( 'P.S. We don\'t like to nag. This message won\'t be shown again.', 'block-lab' ); ?></p>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Edit Your First Block message.
+	 */
+	public function show_edit_block_notice() {
+		$example_post_id = get_option( 'block_lab_example_post_id' );
+
+		if ( ! $example_post_id ) {
+			return;
+		}
+		?>
+		<div class="block-lab-edit-block block-lab-notice notice">
+			<h2>👩‍🔬 <?php echo esc_html_e( 'Ready to begin?', 'block-lab' ); ?></h2>
+			<p class="intro">
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> html tags.
+						__( 'We created this %1$sExample Block%2$s to show you just how easy it is to get started.', 'block-lab' ),
+						'<strong>',
+						'</strong>'
+					)
+				);
+				?>
+			</p>
+			<p>
+				<?php
+				echo wp_kses_post(
+					sprintf(
+						// translators: Placeholders are <strong> and <a> html tags.
+						__( 'You can %1$sEdit%2$s the block to learn more, or just %3$sTrash%4$s it to dismiss this message.', 'block-lab' ),
+						'<strong>',
+						'</strong>',
+						'<a href="' . get_delete_post_link( $example_post_id ) . '" class="trash">',
+						'</a>'
+					)
+				);
+				?>
+			</p>
+			<span class="pointer">☝️</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_add_fields_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
+
+		/**
+		 * We add 4 fields to our Example Block in add_dummy_data().
+		 */
+		if ( count( $block->fields ) > 4 ) {
+			return;
+		}
+		?>
+		<div class="block-lab-add-fields block-lab-notice">
+			<h2>🧐 <?php esc_html_e( 'Try adding a field.', 'block-lab' ); ?></h2>
+			<p><?php esc_html_e( 'Fields let you define the options you see when adding your block to a post.', 'block-lab' ); ?></p>
+			<span class="pointer">👈</span>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Render the Add Fields message.
+	 */
+	public function show_publish_notice() {
+		$post  = get_post();
+		$block = new Block( $post->ID );
```

Comment:
There's nothing wrong with this, but maybe using [get_the_ID()](https://developer.wordpress.org/reference/functions/get_the_ID/) here would be simpler:

```php
$block = new Block( get_the_ID() );
```

`